pwiz_aux/msrc/utility/vendor_api/Waters/WatersRawFile.hpp

```diff
@@ -135,12 +136,28 @@ struct PWIZ_API_DECL RawData
         // For functions over 100, the names become _FUNC0100.DAT
         // Keep track of the maximum function number
         string functionPathmask = rawpath + "/_FUNC*.DAT";
+        string chromatogramPathmask = rawpath + "/_CHRO*.DAT";
         vector<bfs::path> functionFilepaths;
+        vector<bfs::path> chromatogramFilepaths;
         expand_pathmask(functionPathmask, functionFilepaths);
+        expand_pathmask(chromatogramPathmask, chromatogramFilepaths);
         map<int, bfs::path> functionFilepathByNumber;
+        // If there are any chromatograms, then assume that any function without a matching chromatogram is lockmass data and not to be used
+        bool useChromatogramsToDetectLockmassFunction =
+            chromatogramFilepaths.size() > 0 && chromatogramFilepaths.size() == functionFilepaths.size() - 1;
         for (size_t i=0; i < functionFilepaths.size(); ++i)
         {
             string fileName = BFS_STRING(functionFilepaths[i].filename());
+            if (useChromatogramsToDetectLockmassFunction)
+            {
+                string chromatogramFileName = fileName;
+                boost::algorithm::replace_all(chromatogramFileName, "_FUNC", "_CHRO");
+                if (boost::range::find_if(chromatogramFilepaths, 
+                    [chromatogramFileName](const bfs::path& cfp) { return BFS_STRING(cfp.filename()) == chromatogramFileName; }) == chromatogramFilepaths.end())
+                {
+                    continue; // No matching chromatogram - presumably the lockmass function, don't add to list

```

Comment:
This is going to make the functions entirely disappear from pwiz? I'm not sure that's a good idea unless you want to make it a Reader::Config option. It's entirely likely that users will want to see their lockmass spectra. Instead I think we could annotate those spectra with a userParam (at this point in the code it would need to be a new API function like `IsFunctionLockmass(int functionNumber)` which SpectrumList_Waters would call) and clients could exclude the lockmass spectra that way. But the Reader::Config route is valid too.

Reply:
I'm open to better ways to use this information. Though I thought msconvert already was trying to not pass lockmass scans through (Skyline is certainly trying to ignore them).

Is there any likelihood that a function lacks a matching chromatogram for reasons other than being lockmass data?

Reply:
It's only Skyline that currently has the heuristic to exclude lockmass functions. I have no idea about the CHRO files. How did you come up with this heuristic? Have you checked PDA functions? If not I'll have to sleuth around to remember which of my test files has PDA.

Reply:
Photo Diode Array. Records wavelength vs. intensity spectra (which can have wavelength vs intensity chromatograms, but I don't know if that maps to CHRO files). Skyline ignores such spectra of course but I've definitely had users that use them.

Reply:
Yeah, I'll need to make sure that it only considers functions that are mass spec data.

Reply:
Matt, can you take another look? I think I've addressed your concerns.